Scripts/Game/ODIN/Editor/Components/EditableEntity/SCR_EditableGroupComponent.c

```diff
@@ -33,6 +24,23 @@ modded class SCR_EditableGroupComponent : SCR_EditableEntityComponent
 		ReindexWaypoints();
 	}
 	
+	protected bool ODIN_HasWaypointCycle()
+	{
+		array<AIWaypoint> waypoints = {};
+		m_Group.GetWaypoints(waypoints);
+		
+		foreach (AIWaypoint waypoint : waypoints)
+		{
+				
+			if (AIWaypointCycle.Cast(waypoint))
+			{
+				return true;
+			};
+		};
```

Comment:
I don't think it's actually worth it but technically I would guess iterating through the waypoints backwards would more often find the cycle waypoint faster, as it would often be the last waypoint added to a group?

But unless someone starts putting 10,000+ waypoints down, it's likely a negligible difference.